docs/rune-modelling-component.md

```diff
@@ -912,31 +912,69 @@ Rune supports basic arithmetic operators
 
 The `default` operator takes two values of matching type. If the value to the left of the default is empty then the value of to the right of the default will be returned. Note that the type and cardinality of both sides of the operator must match for the syntax to be valid.
 
-### Switch Operator
+#### Switch Operator
 
 The `switch` operator takes as its left hand input an argument on which to perform case analysis. The right side of the operator takes a set of case statements which define a return value for the expression when matching that case to the input.
 
 ```Haskell
    "valueB" switch
-      "valueA" then "resultA"
-      "valueB" then "resultB"
+      "valueA" then "resultA",
+      "valueB" then "resultB",
       default "resultC"
 ```
 
 The `switch` operator can also operate over enumerations and in the case where all enumeration values are not provided as case statements then a syntax validation error will occur until either all enumeration values or a default value is provided.
 
 ```Haskell
-  "aCondition" switch 
-    "aCondition" then SomeEnum -> A,
-    "bCondition" then SomeEnum -> B,
-    "cCondition" then SomeEnum -> C,
-    default SomeEnum -> D
+  enumInput switch 
+    A then "result A",
+    B then "result B",
+    C then "result C",
+    default "other"
 ```
 
 {{< notice info "Note" >}}
-The `default` case is optional, in case when there is no match then the `empty` value is returned.
+The `default` case is optional. In case when there is no match then the `empty` value is returned.
 {{< /notice >}}
 
+##### `switch` for `choice` types
+
+Consider the following model for a `Vehicle`.
```

Comment:
I find this example overly complex and therefore difficult to understand.

Could you simplify the example to show a non-nested and syntactically correct use.  For example:

```
choice Vehicle:
  PetrolCar
  ElectricCar

type Car:
  fuelCapacity number (1..1)

type ElectricCar:
  batteryCapacity number (1..1)
```

and then a simple use, eg:

```
set efficiency:  vehicle switch 
  ElectricCar        then 5 * batteryCapacity // assume 5 kilometres per kWh of battery
  Car                then 15 * fuelCapacity   // assume 15 kilometres per litre of fuel
```